src/ProxyManager/Generator/ClassGenerator.php

```diff
@@ -33,7 +33,7 @@ class ClassGenerator extends ZendClassGenerator
     /**
      * {@inheritDoc}
      */
-    public function setExtendedClass($extendedClass) : self
+    public function setExtendedClass($extendedClass) : parent
```

Comment:
Why a `set` method is returning something? seems pretty weird. 

Note: I see the context.